acdc/main.py

```diff
@@ -281,18 +266,25 @@
     if i==0:
         exp.save_edges("edges.pkl")
 
-    if TASK in ["docstring", "induction"]:
-        with torch.no_grad():
-            test_metric_values = {}
-            for k, fn in test_metric_fns.items():
-                test_metric_values["test_"+k] = fn(exp.model(test_metric_data))
-        if USING_WANDB:
-            wandb.log(test_metric_values)
-
     if exp.current_node is None or SINGLE_STEP:
         break
 
-exp.save_edges("another_final_edges.pkl") 
+exp.save_edges("another_final_edges.pkl")
+
+
+# Save test dataset
+
+with torch.no_grad():
+    exp.ref_ds = things.test_patch_data
+    exp.model.reset_hooks()
```

Comment:
I don't think you want to reset hooks and setup the second cache. This will leave the model with no hooks on it at all, so `exp.model(things.test_data)` will be the same as `tl_model(things.test_data)` on a fresh `HookedTransformer`. I think you just want to delete these two lines

Reply:
We talked about this, I thought re-running `.setup_second_cache()` was the way to run ACDC on a test set. It seems to output sensible things.

Also it internally calls `.reset_hooks()` so it's almost certainly fine to run it before. I can delete it because it's superfluous but I'm *not* going to delete `setup_second_cache` after `exp.ref_ds = things.test_patch_data` unless you make a better case.

Reply:
How's the new change @ArthurConmy ?